nvbench/CMakeLists.txt

```
@@ -132,12 +132,20 @@ endif()
 # Use `nvbench_git_revision` but don't expose it as part of our export information
 # In CMake 3.27 this is easily done with $<COMPILE_ONLY>, before that we have to
 # manual copy the include path
+#
+# The call to `rapids_cmake_write_git_revision_file` must be in the same
+# CMakeLists.txt as the consumer ( nvbench ) for CMake to get the dependency
+# graph correct.
+rapids_cmake_write_git_revision_file(
+    nvbench_git_revision
+    "${NVBench_BINARY_DIR}/nvbench/detail/git_revision.cuh"
+  )
```

Nitpick, some whitespace damage
```suggestion
  nvbench_git_revision
  "${NVBench_BINARY_DIR}/nvbench/detail/git_revision.cuh"
)
```